compiler/lib/driver.ml

```diff
@@ -677,56 +679,59 @@ let full ~standalone ~wrap_with_fun ~profile ~link ~source_map formatter d p =
        | O3 -> o3)
     +> exact_calls ~deadcode_sentinal profile
     +> effects ~deadcode_sentinal
-    +> map_fst (if Config.Flag.effects () then fun x -> x else Generate_closure.f)
+    +> map_fst
+         (match target with
+         | `JavaScript -> if Config.Flag.effects () then Fun.id else Generate_closure.f
+         | `Wasm -> Fun.id)
     +> map_fst deadcode'
   in
-  let emit =
+  if times () then Format.eprintf "Start Optimizing...@.";
+  let t = Timer.make () in
+  let r = opt p in
+  let () = if times () then Format.eprintf " optimizations : %a@." Timer.print t in
+  r
+
+let full ~standalone ~wrap_with_fun ~profile ~link ~source_map ~formatter d p =
+  let deadcode_sentinal =
+    (* If deadcode is disabled, this field is just fresh variable *)
+    Code.Var.fresh_n "undef"
+  in
```

Comment:
This should be in function `optimize`.

Reply:
Are you sure? It’s used as an argument to generate below, as well.

Reply:
Maybe it should be returned by `optimize`, then.

Reply:
Done